### PR TITLE
Support custom scalars as variables in an operation

### PR DESCRIPTION
**Pull request by jaredly**
## Summary:
Found when trying to run on all webapp operations https://khanacademy.slack.com/archives/CEA6W0F6F/p1648242239192519?thread_ts=1648130416.886269&cid=CEA6W0F6F

Issue: https://khanacademy.atlassian.net/browse/FEI-4411

## Test plan:
`yarn jest`!

### CODE REVIEW

File: Readme.md [hunk 1]
```diff
@@ -94,10 +94,11 @@ Then you'll want to make a pseudo-'test' that makes sure to 'require' all of the
 jest will process them and our mock will see them. 
 ```js
 // generate-types.test.js
-import {findFilesWithQueries} from '../tools/graphql-flow/find-files-with-gql';
+import {findGraphqlTagReferences} from '../tools/graphql-flow/find-files-with-gql';
+import path from 'path';
 
 if (process.env.GRAPHQL_FLOW) {
-    findFilesWithQueries(path.join(__dirname, '..')).forEach(name => {
+    findGraphqlTagReferences(path.join(__dirname, '..')).forEach(name => {
```
Comment (jaredly) [inline Readme.md:101]: some readme things that I realized were out of date

File: package.json [hunk 2]
```diff
@@ -11,30 +11,30 @@
...
         "@babel/generator": "^7.17.3",
-        "@babel/polyfill": "^7.0.0",
-        "@babel/preset-env": "^7.16.11",
-        "@babel/preset-flow": "^7.16.7",
         "@babel/types": "^7.17.0",
         "apollo-utilities": "^1.3.4",
-        "babel-jest": "23.4.2",
-        "eslint-plugin-flowtype": "^8.0.3",
         "graphql": "14.2.1",
-        "graphql-tag": "2.10.1",
```
Comment (jaredly) [inline package.json:37]: a bunch of dependencies that aren't needed outside of dev

File: src/jest-mock-graphql-tag.js [hunk 3]
```diff
@@ -104,24 +104,26 @@ type SpyOptions = {
...
+ *     return spyOnGraphqlTagToCollectQueries(
+ *         jest.requireActual('graphql-tag'),
+ *         introspectionData,
+ *     );
+ * });
+ *
+ * If both pragma and loosePragma are empty, then all graphql
+ * documents will be processed. Otherwise, only documents
+ * with one of the pragmas will be processed.
+ */
```
Comment (jaredly) [inline src/jest-mock-graphql-tag.js:126]: vscode likes these better

File: .changeset/ninety-avocados-act.md [hunk 4]
```diff
@@ -0,0 +1,5 @@
+---
+'@khanacademy/graphql-flow': patch
+---
+
+Support custom scalars as variables in an operation
```
Comment (benchristel) [inline .changeset/ninety-avocados-act.md:5]: Why is this file called `ninety-avocados-act`?

**Review by benchristel — APPROVED** [review]
LGTM!

File: .changeset/ninety-avocados-act.md (hunk 4 above)
Comment (jaredly) [reply .changeset/ninety-avocados-act.md:5]: the changeset has this funny way of naming files. I'm not really sure what the point is

**Pull request merged**